lib/workload/stateless/stacks/bclconvert-manager/translator_service/icav2_event_translator.py

```diff
@@ -0,0 +1,504 @@
+            ]
+        )
+        logger.info(f"Internal event sent to the event bus.")
+    except Exception as e:
+        raise Exception("Failed to send event to the event bus. Error: ", e)
+
+
+# Store the internal event in the DynamoDB table
+def store_events_into_dynamodb(internal_ica_event, table_name, event_details) -> None:
+    try:
+        db_uuid = generate_db_uuid(None, None).get("db_uuid",'')
+        dynamodb.put_item(
+            TableName=table_name,
+            Item={
+                'id': {'S': db_uuid},
+                'id_type': {'S': 'db_uuid'},
+                'analysis_id': {'S': internal_ica_event.detail.payload.get('data', '').get("analysisId", '')},
+                'analysis_status': {'S': internal_ica_event.detail.status},
+                # 'SUCCEEDED', 'FAILED', 'INPROGRESS', 'ABORTED', 'UNKNOWN
+                "portal_run_id": {'S': internal_ica_event.detail.portalRunId},
+                'original_external_event': {'S': json.dumps(event_details)},
+                'translated_internal_ica_event': {'S': json.dumps(internal_ica_event.detail.to_dict())},
+                'timestamp': {'S': internal_ica_event.detail.timestamp}
+            }
+        )
+        logger.info(f"Original and Internal events stored in the DynamoDB table.")
+
+        # update dynamodb table with the new generate db_uuid
+        dynamodb.update_item(
+            TableName=table_name,
+            Key={
+                'id': {'S': internal_ica_event.detail.payload.get("data", '').get("analysisId", '')},
+                'id_type': {'S': 'analysis_id'}
+            },
+            UpdateExpression='SET db_uuid = :db_uuid',
+            ExpressionAttributeValues={
+                ':db_uuid': {'S': db_uuid}
+            }
+        )
+        dynamodb.update_item(
+            TableName=table_name,
+            Key={
+                'id': {'S': internal_ica_event.detail.portalRunId},
+                'id_type': {'S': 'portal_run_id'}
+            },
+            UpdateExpression='SET db_uuid = :db_uuid',
+            ExpressionAttributeValues={
+                ':db_uuid': {'S': db_uuid}
+            }
+        )
+        logger.info(f"db_uuid updated in anaylsis_id and portal_run_id record.")
+    except Exception as e:
+        raise Exception("Failed to store event in the DynamoDB table. Error: ", e)
+
+
+# Convert from Entity model to aws event object with aws event envelope
+def translate_to_aws_event(event) -> AWSEvent:
+    return AWSEvent(
+        detail=get_event_details(event),
+        detail_type="WorkflowRunStateChange",
+        source="orcabus.bclconvertmanager",
+        # version="0.1.0",  # comment as the version is managed by the evnet bus
+    )
+
+
+# Convert from entity module to internal event details
+def get_event_details(event) -> WorkflowRunStateChange:
+    # Extract relevant fields from the event payload
+    analysis_status = event.get("eventParameters", {}).get("analysisStatus", '')
```

Comment:
Would "UNSPECIFIED" be a better default?

Reply:
sure, added